roscala/src/main/scala/coop/rchain/rosette/Tuple.scala

```diff
@@ -7,12 +7,30 @@ package coop.rchain.rosette
   *  in messages, and as a primitive data structure in the definition
   *  of other actors
   */
+sealed trait TupleError
+case object AbsentRest extends TupleError
+case object InvalidRest extends TupleError
+
 case class Tuple(elem: Seq[Ob],
                  override val parent: Ob,
                  override val meta: Ob,
                  override val slot: Seq[Ob])
     extends Ob {
 
+  def accepts(msg: Ctxt): Boolean =
+    if (this == Tuple.NIL) {
+      true
+    } else {
+      this.elem.exists(_.matches(msg))
+    }
+
+  def flattenRest(): Either[TupleError, Tuple] =
```

Comment:
The name seems strange to me; maybe flattenLast instead?

This seems strange to me.  Suppose we have `this = [9, [1, 2, 3], [7, 6, 5]]`.  Then we'd return `Right([9, [1, 2, 3], [7, 6, 5], 7, 6, 5])`.  Is that what you intended?  If so, it's not really flattening but rather concatenating, so I'd name it `concatLast` or something.

Reply:
I named it `flattenRest` because it seems that in Rosette you call the last element `rest` (if the last element is a `Tuple`). `flattenLast` would be less precise in my opinion.

The result I am looking for with regards to your example is: `Right([9, [1, 2, 3], 7, 6, 5])`.
So the current implementation is indeed not correct. Thanks for pointing that out.